httpclient5/src/main/java/org/apache/hc/client5/http/config/RequestConfig.java

```diff
@@ -43,7 +43,6 @@
 public class RequestConfig implements Cloneable {
 
     private static final Timeout DEFAULT_CONNECTION_REQUEST_TIMEOUT = Timeout.ofMinutes(3);
```

Comment:
Yet another stupid question: Since this is the time how long the pool at most should require to return a working connection, why is this a HTTP request level config?

Reply:
@michael-o This really depends how one looks at it. This setting is used by each and every request and as such makes sense at the request level. However, indeed, the inner machinery is entirely in the connection manager's responsibility. 
I initially moved that parameter to `ConnectionConfig` but if I remember it correctly you questioned that decision. 
Basically as far as I am concerned it can be either way. Blonds or Brunettes, so to speak. However I tend to see it more of a request level setting.

Reply:
Let's leave it as-is, but I still think that a connection lease timeout should be on a pool. See here: https://github.com/apache/httpcomponents-client/pull/296#discussion_r597967488